### DISCUSSION TITLE
Process all gomplate.yaml files in a directory

### DISCUSSION THREAD

**Discussion by theskipster** [hairyhenderson/gomplate#1258] [Q&A] [answered]
Is there a way, from the command line, to process all gomplate.yaml files in a directory?  I was hoping to have an automated process to be able to just add more configs and have them automatically generate the output files.

**Comment by hairyhenderson** [comment, marked as answer]
By `gomplate.yaml` I assume you mean the `.gomplate.yaml` [config file](https://docs.gomplate.ca/config/)? The intent is to have a single config file for a single gomplate execution. You could probably script it like:

```bash
for config in mydir/*.yaml; do
  gomplate --config $config
done
```

If you want to process a number of templates in the same execution, you could use `--input-dir` (or `inputDir` in the config) - see https://docs.gomplate.ca/config/#inputdir. The advantage of doing that is you can share datasources/contexts. But that's obviously not always desired 😉.

Please let me know if this answers the question - if not, let me know what's different about your specific use-case!
